spark/src/main/java/org/apache/iceberg/spark/source/DeleteRowReader.java

```diff
@@ -31,13 +32,16 @@
 import org.apache.spark.rdd.InputFileBlockHolder;
 import org.apache.spark.sql.catalyst.InternalRow;
 
-public class EqualityDeleteRowReader extends RowDataReader {
+public class DeleteRowReader extends RowDataReader {
   private final Schema expectedSchema;
+  private final FileContent deleteContent;
 
-  public EqualityDeleteRowReader(CombinedScanTask task, Schema schema, Schema expectedSchema, String nameMapping,
-                                 FileIO io, EncryptionManager encryptionManager, boolean caseSensitive) {
+  public DeleteRowReader(CombinedScanTask task, Schema schema, Schema expectedSchema, String nameMapping,
+                         FileIO io, EncryptionManager encryptionManager, boolean caseSensitive,
+                         FileContent deleteContent) {
```

Comment:
There is no guarantee that the `CombinedScanTask` doesn't have both position and equality deletes to apply, so it doesn't make sense to add this argument and handle just one. I think that this reader should return all deleted rows from a file, no matter which kind of delete was encoded. Right?

Reply:
Yes, there is no guarantee about what kind of deletes a `CombinedScanTask` contains. Here I'd like to expose the option to the user to select one kind of delete to rewrite. How about returns all deleted rows when `deleteContent` is not passed? That should be a valid option I think.